src/gluonnlp/embedding/evaluation.py

```diff
@@ -0,0 +1,249 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=eval-used, redefined-outer-name
+"""Functions for intrinsic and extrinsic word embedding evaluation"""
+
+import mxnet as mx
+import numpy as np
+from ..op import l2_normalize
+
+__all__ = ['CosineSimilarity', 'ThreeCosMul', 'ThreeCosAdd', 'HyperbolicCosineSimilarity']
```

Comment:
We can add these in a `scripts/` folder as part of the evaluation scripts for the similarity and analogy datasets. We don't need to add it to the main API as part of this PR. (The `scripts/` for evaluation are not required for automl and you don't need to add it if you're not interested in trying it)